src/e2e-test/java/io/cdap/plugin/pubsub/stepsdesign/PubSubConnector.java

```diff
@@ -0,0 +1,401 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.pubsub.stepsdesign;
+
+import io.cdap.e2e.pages.actions.CdfBigQueryPropertiesActions;
+import io.cdap.e2e.pages.actions.CdfGcsActions;
+import io.cdap.e2e.pages.actions.CdfLogActions;
+import io.cdap.e2e.pages.actions.CdfPipelineRunAction;
+import io.cdap.e2e.pages.actions.CdfStudioActions;
+import io.cdap.e2e.pages.locators.CdfGCSLocators;
+import io.cdap.e2e.pages.locators.CdfStudioLocators;
+import io.cdap.e2e.utils.CdfHelper;
+import io.cdap.e2e.utils.GcpClient;
+import io.cdap.e2e.utils.SeleniumDriver;
+import io.cdap.e2e.utils.SeleniumHelper;
+import io.cdap.plugin.pubsub.actions.PubSubActions;
+import io.cdap.plugin.pubsub.locators.PubSubLocators;
+import io.cdap.plugin.utils.E2ETestUtils;
+import io.cucumber.java.en.Given;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.Assert;
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import stepsdesign.BeforeActions;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_COLOR;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_PUBSUB_DELAY_THRESHOLD;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_PUBSUB_ERROR_THRESHOLD;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_PUBSUB_INVALID_ADVANCED_FIELDS;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_PUBSUB_MAX_BATCH_COUNT;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_PUBSUB_MAX_BATCH_SIZE;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_PUBSUB_RETRY_TIMEOUT;
+import static io.cdap.plugin.utils.E2ETestConstants.ERROR_MSG_VALIDATION;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * PubSub connector related step design.
+ */
+
+public class PubSubConnector implements CdfHelper {
+  List<String> propertiesOutputSchema = new ArrayList<String>();
```

Comment:
schema is a map

Reply:
Renamed... This is schema column names list.

Reply:
Schema contains a column names and data type for each column. You need to verify that all the fields are present in the output schema with the correct type.

Reply:
Schema verification with Column name and data type is added